CMake/build.cmake

```diff
@@ -13,23 +13,27 @@ if(NOT CMAKE_BUILD_TYPE)
   set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
+# SENSEI should always be position independent
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+# SENSEI requires minimum C++11
+if (NOT CMAKE_CXX_STANDARD OR CMAKE_CXX_STANDARD LESS 11)
+  set(CMAKE_CXX_STANDARD 11)
+endif ()
+
 if (NOT MSVC)
-  if (NOT CMAKE_CXX_FLAGS)
-  set(tmp "-fPIC -std=c++11 -Wall -Wextra")
```

Comment:
@kwryankrattiger Is this addressing a specific issue? If so could in that case the user not explicitly set `CMAKE_CXX_FLAGS` directly as they need?

We generally should use `-fPIC` so that SENSEI can be linked against apps that use dynamic linking. Also, my recollection of this is that without PIC, there are link errors when static linking and Python are both enabled.

Our strategy has been to only set it if the user has not otherwise set `CMAKE_CXX_FLAGS` hence if the user does not want PIC they can set `CMAKE_CXX_FLAGS`. I think we should keep -fPIC and let the user override as needed via explicit CMAKE_CXX_FLAGS. Why not?